src/components/pages/user/edit/tabs/app-user/app-user-form.tsx

```diff
@@ -0,0 +1,54 @@
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { Button } from "@chakra-ui/react";
+import { SelectInputField } from "@components/form/select";
+import { yupResolver } from "@hookform/resolvers/yup";
+import notification from "@utils/notification";
+import useTranslation from "next-translate/useTranslation";
+import React from "react";
+import { FormProvider, useForm } from "react-hook-form";
+import * as Yup from "yup";
+
+export default function AppAppUser({ setIsCreate, projectList, setProjectId }) {
+  const { t } = useTranslation();
+
+  const projectForm = useForm<any>({
+    mode: "onBlur",
+    resolver: yupResolver(
+      Yup.object().shape({
+        projectId: Yup.string().required(),
+      })
+    ),
+  });
+
+  const handleFormSubmit = async () => {
+    if (!projectForm.getValues("projectId")) {
+      notification(t("common:action.project_select"));
+      return;
+    }
+    setProjectId(projectForm.getValues("projectId"));
+  };
+
+  return (
+    <FormProvider {...projectForm}>
+      <form className="fade">
+        <Button
+          mb={4}
+          type="button"
+          size="sm"
+          onClick={() => setIsCreate(false)}
+          leftIcon={<ArrowBackIcon />}
+        >
+          {t("common:prev")}
+        </Button>
+        <SelectInputField
+          isRequired={true}
+          name="projectId"
+          options={projectList.map((item) => ({ label: item.name, value: item.id }))}
```

Comment:
Extract this map expression in a separate array called `projectListOptions` and 
put change it to `options=projectListOptions`.
pros of doing it:
1)Better readability of code by other developers.
2)Better maintainability of code.
3)Makes debugging easier